akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -317,6 +318,96 @@ class CircuitBreaker(
   }
 
   /**
+   * Adds a callback to execute when call finished with success.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Long ⇒ Unit): CircuitBreaker = onCallSuccess(new Consumer[Long] {
```

Comment:
I think that it's fine to have `Long` in nanoseconds as the API. It's fairly common for monitoring APIs to try to minimize the number of unnecessary allocations that are done, and have timings in nano seconds.